tsl/src/continuous_aggs/refresh.c

```diff
@@ -764,9 +765,16 @@ continuous_agg_refresh_internal(const ContinuousAgg *cagg,
 	int32 mat_id = cagg->data.mat_hypertable_id;
 	InternalTimeRange refresh_window = *refresh_window_arg;
 	int64 invalidation_threshold;
+	bool nonatomic = ts_process_utility_is_context_nonatomic();
+
+	/* Reset the saved ProcessUtilityContext value promptly before
+	 * calling PreventCommandIfReadOnly so the potential unsupported
+	 * (atomic) value won't linger there in case of ereport exit.
+	 */
+	ts_process_utility_context_reset();
```

Comment:
Why don't error out here when nonatomic is false?

Reply:
Moved `Prevent*` checks before the SPI call.